node/src/rpc/rpc_reducer.rs

```diff
@@ -1,114 +1,395 @@
                     return;
                 };
                 rpc.status = RpcRequestStatus::Success { time: meta.time() };
+                let dispatcher = state_context.into_dispatcher();
+                dispatcher
+                    .push(RpcEffectfulAction::P2pConnectionIncomingSuccess { rpc_id: *rpc_id });
             }
             RpcAction::ScanStateSummaryGetInit { rpc_id, query } => {
                 let rpc_state = RpcRequestState {
                     req: RpcRequest::ScanStateSummaryGet(query.clone()),
                     status: RpcRequestStatus::Init { time: meta.time() },
                     data: Default::default(),
                 };
-                self.requests.insert(*rpc_id, rpc_state);
+                state.requests.insert(*rpc_id, rpc_state);
+
+                let dispatcher = state_context.into_dispatcher();
+                dispatcher.push(RpcAction::ScanStateSummaryLedgerGetInit { rpc_id: *rpc_id });
+            }
+            RpcAction::ScanStateSummaryLedgerGetInit { rpc_id } => {
+                let (dispatcher, state) = state_context.into_dispatcher_and_state();
+                let transition_frontier = &state.transition_frontier;
+
+                let Some(query) = None.or_else(|| {
+                    let req = state.rpc.requests.get(rpc_id)?;
+                    match &req.req {
+                        RpcRequest::ScanStateSummaryGet(query) => Some(query),
+                        _ => None,
+                    }
+                }) else {
+                    return;
+                };
+
+                let block = match query {
+                    RpcScanStateSummaryGetQuery::ForBestTip => {
+                        transition_frontier.best_tip_breadcrumb()
+                    }
+                    RpcScanStateSummaryGetQuery::ForBlockWithHash(hash) => transition_frontier
+                        .best_chain
+                        .iter()
+                        .rev()
+                        .find(|b| b.hash() == hash),
+                    RpcScanStateSummaryGetQuery::ForBlockWithHeight(height) => transition_frontier
+                        .best_chain
+                        .iter()
+                        .rev()
+                        .find(|b| b.height() == *height),
+                };
+                let block = match block {
+                    Some(v) => v.clone(),
+                    None => {
+                        dispatcher.push(RpcAction::ScanStateSummaryGetPending {
+                            rpc_id: *rpc_id,
+                            block: None,
+                        });
+                        dispatcher.push(RpcAction::ScanStateSummaryGetSuccess {
+                            rpc_id: *rpc_id,
+                            scan_state: Ok(Vec::new()),
+                        });
+                        return;
+                    }
+                };
+
+                // TODO: add callbacks here
```

Comment:
May be a good time to solve this `TODO`. Pattern is the same as the `on_init` callback in other actions. See `Callbacks instead of conditional dispatches` in the architecture document.